src/com/projectkorra/projectkorra/ability/util/MultiAbilityManager.java

```diff
@@ -180,44 +178,33 @@ public static boolean canChangeSlot(final Player player, int slot) {
 	 * @param player
 	 */
 	public static void unbindMultiAbility(final Player player) {
-		if (playerAbilities.containsKey(player)) {
-			final HashMap<Integer, String> prevBinds = playerAbilities.get(player);
-			final BendingPlayer bPlayer = BendingPlayer.getBendingPlayer(player);
-			if (bPlayer == null) {
-				return;
-			}
-
-			int lastNonNull = -1;
-			for (int i = 1; i < 10; i++) {
-				if (prevBinds.get(i) != null) {
-					lastNonNull = i;
-				}
-				bPlayer.getAbilities().put(i, prevBinds.get(i));
-			}
-			if (lastNonNull > -1) {
-				GeneralMethods.saveAbility(bPlayer, lastNonNull, prevBinds.get(lastNonNull));
-			}
-
-			if (player.isOnline()) {
-				bPlayer.addCooldown("MAM_Setup", 1L); // Support for bending scoreboards.
-			}
-			playerAbilities.remove(player);
+		if (!player.isOnline()) {
+			return;
 		}
-
-		if (playerSlot.containsKey(player)) {
-			if (player.isOnline()) {
-				player.getInventory().setHeldItemSlot(playerSlot.get(player));
-			}
-			playerSlot.remove(player);
-		} else {
-			if (player.isOnline()) {
-				player.getInventory().setHeldItemSlot(0);
-			}
+		
+		playerAbilities.compute(player, MultiAbilityManager::resetBinds);
+		playerBoundAbility.remove(player);
+		playerSlot.remove(player);
+	}
+	
+	private static HashMap<Integer, String> resetBinds(Player player, HashMap<Integer, String> prevBinds) {
+		if (prevBinds == null) {
+			return null;
+		}
+		
+		final BendingPlayer bPlayer = BendingPlayer.getBendingPlayer(player);
+		if (bPlayer == null) {
+			return null;
 		}
+		
+		player.getInventory().setHeldItemSlot(playerSlot.getOrDefault(player, 0));
+		ProjectKorra.plugin.getServer().getPluginManager().callEvent(new PlayerBindChangeEvent(player, playerBoundAbility.get(player), false, true));
 
-		if (playerBoundAbility.containsKey(player)) {
-			playerBoundAbility.remove(player);
+		for (int i = 1; i < 10; i++) {
+			bPlayer.getAbilities().put(i, prevBinds.get(i));
 		}
+		
+		return null;
```

Comment:
Why do we just have a bunch of return nulls here, just make it a void type so people dont get confused, or at least return a boolean to see whether it was succesful.

Reply:
It's private so nobody will be using it, and it returns null because it's used above as the function for `Map#computeIfPresent` method called. We want to remove the mapping from the map so the function needs to return null after reseting the player's binds to the stored binds.